Content.Shared/Clothing/LoadoutSystem.cs

```diff
@@ -24,12 +25,292 @@ public override void Initialize()
         SubscribeLocalEvent<LoadoutComponent, MapInitEvent>(OnMapInit);
     }
 
+
     private void OnMapInit(EntityUid uid, LoadoutComponent component, MapInitEvent args)
     {
         if (component.Prototypes == null)
             return;
 
-        var proto = _protoMan.Index<StartingGearPrototype>(_random.Pick(component.Prototypes));
+        var proto = _prototype.Index<StartingGearPrototype>(_random.Pick(component.Prototypes));
         _station.EquipStartingGear(uid, proto, null);
     }
+
+
+    /// <inheritdoc cref="ApplyCharacterLoadout(Robust.Shared.GameObjects.EntityUid,string,Content.Shared.Preferences.HumanoidCharacterProfile,System.Collections.Generic.Dictionary{string,System.TimeSpan}?)"/>
+    public List<EntityUid> ApplyCharacterLoadout(EntityUid uid, string job, HumanoidCharacterProfile profile, Dictionary<string, TimeSpan>? playTimes = null)
+    {
+        var jobPrototype = _prototype.Index<JobPrototype>(job);
+        return ApplyCharacterLoadout(uid, jobPrototype, profile, playTimes);
+    }
+
+    /// <summary>
+    ///     Equips entities from a <see cref="HumanoidCharacterProfile"/>'s loadout preferences to a given entity
+    /// </summary>
+    /// <param name="uid">The entity to give the loadout items to</param>
+    /// <param name="job">The job to use for loadout whitelist/blacklist (should be the job of the entity)</param>
+    /// <param name="profile">The profile to get loadout items from (should be the entity's, or at least have the same species as the entity)</param>
+    /// <param name="playTimes">Playtime for the player for use with playtime requirements</param>
+    /// <returns>A list of loadout items that couldn't be equipped but passed checks</returns>
+    public List<EntityUid> ApplyCharacterLoadout(EntityUid uid, JobPrototype job, HumanoidCharacterProfile profile, Dictionary<string, TimeSpan>? playTimes = null)
+    {
+        var failedLoadouts = new List<EntityUid>();
+
+        foreach (var loadout in profile.LoadoutPreferences)
```

Comment:
As below rather than hardcoding in every single prototype as separate fields just make the requirements a list of <jobrequirement / speciesrequirement / pointcost> instead. That way any forks that don't want stuff like for example point cost can trivially remove it without having to make a bunch of code changes.